File: app/src/main/java/org/growingstems/scouting/TransparentImageButton.java
package org.growingstems.scouting;

import android.annotation.SuppressLint;
import android.content.Context;
import android.graphics.Color;
import android.graphics.drawable.BitmapDrawable;
import android.graphics.drawable.Drawable;
import android.graphics.drawable.VectorDrawable;
import android.util.AttributeSet;
import android.view.MotionEvent;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

public class TransparentImageButton extends androidx.appcompat.widget.AppCompatImageButton {



	public TransparentImageButton(@NonNull Context context) {
		super(context);
	}

	public TransparentImageButton(@NonNull Context context, @Nullable AttributeSet attrs) {
		super(context, attrs);
	}

	public TransparentImageButton(
			@NonNull Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
		super(context, attrs, defStyleAttr);
	}

	@SuppressLint("ClickableViewAccessibility")
	@Override
	public boolean onTouchEvent(MotionEvent event) {
		Drawable d = getDrawable();

		if (d instanceof BitmapDrawable) {
			if (isPixelTransparent((BitmapDrawable)d, (int)event.getX(), (int)event.getY())) {
				return false;
			} else {
				return super.onTouchEvent(event);
			}
		} else if (d instanceof VectorDrawable) {
			if (isPixelTransparent((VectorDrawable)d, (int)event.getX(), (int)event.getY())) {
				return false;
			} else {
				return super.onTouchEvent(event);
			}
		}
		return super.onTouchEvent(event);
	}

	private boolean isPixelTransparent(BitmapDrawable d, int x, int y) {

		int color = Color.TRANSPARENT;
		try {
			color = d.getBitmap().getPixel(x, y);
		} catch (Exception e) {
			//TODO
		}

		return color == Color.TRANSPARENT;
	}

	private boolean isPixelTransparent(VectorDrawable d, int x, int y) {

		int color = Color.TRANSPARENT;
		try {
			boolean disableCache = false;
			if (!isDrawingCacheEnabled()) {
				setDrawingCacheEnabled(true);
				disableCache = true;
			}
			color = getDrawingCache().getPixel(x, y);
			if (disableCache) {
				setDrawingCacheEnabled(false);
			}
		} catch (Exception e) {
			//TODO
		}

		return color == Color.TRANSPARENT;
	}
}

File: app/src/main/java/org/growingstems/scouting/TransparentImageButton.kt
package org.growingstems.scouting

import android.annotation.SuppressLint
import android.content.Context
import android.graphics.Color
import android.graphics.drawable.Drawable
import android.util.AttributeSet
import android.view.MotionEvent
import androidx.appcompat.widget.AppCompatImageButton
import androidx.core.graphics.drawable.toBitmapOrNull

class TransparentImageButton : AppCompatImageButton {
    constructor(context: Context) : super(context)
    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs)
    constructor(
        context: Context, attrs: AttributeSet?, defStyleAttr: Int
    ) : super(context, attrs, defStyleAttr)

    @SuppressLint("ClickableViewAccessibility")
    override fun onTouchEvent(event: MotionEvent): Boolean {
        val d = drawable
        return if (isPixelTransparent(d, event.x.toInt(), event.y.toInt())) {
            false
        } else {
            super.onTouchEvent(event)
        }
    }

    private fun isPixelTransparent(d: Drawable, x: Int, y: Int): Boolean {
        try {
            d.toBitmapOrNull(width, height, null)?.let {
                return it.getPixel(x, y) == Color.TRANSPARENT
            }
        } catch (e: Exception) {
            //TODO
        }
        return true
    }
}
